network.go

```diff
@@ -235,16 +209,50 @@ func (s *sandbox) updateInterface(netHandle *netlink.Handle, iface *pb.Interface
 	if lAttrs != nil && (lAttrs.Flags&net.FlagUp) == net.FlagUp {
 		// The link is up, makes sure we get it down before
 		// doing any modification.
-		if err := netHandle.LinkSetDown(link); err != nil {
-			return err
+		err = netHandle.LinkSetDown(link)
+		if err != nil {
+			goto error_case
 		}
 	}
 
-	if err := updateLink(netHandle, link, iface, actionType); err != nil {
-		return err
+	err = updateLink(netHandle, link, iface)
+	if err != nil {
+		goto error_case
+	}
+	err = netHandle.LinkSetUp(link)
+
+error_case:
+	// in the event that an error occurred during the interface update, make sure we return
+	// the resulting state instead of the requested state
+	if err != nil {
+		resultingIfc = getInterface(netHandle, link)
+	} else {
+		resultingIfc = iface
+	}
+
+	return resultingIfc, netHandle.LinkSetUp(link)
+}
+
+// getInterface will retrieve interface details from the provided link
+func getInterface(netHandle *netlink.Handle, link netlink.Link) *pb.Interface {
+	var ifc pb.Interface
+	linkAttrs := link.Attrs()
+	ifc.Name = linkAttrs.Name
+	ifc.Mtu = uint64(linkAttrs.MTU)
+	ifc.HwAddr = linkAttrs.HardwareAddr.String()
+
+	addrs, _ := netHandle.AddrList(link, netlink.FAMILY_V4)
```

Comment:
Why are we not checking the returned error value? I'm guessing because this function doesn't return an error (because this function is called in an error scenario). But I'd still check the error and log it atleast.

Reply:
Sure - will do.